### ISSUE TITLE
Make Scheduler RecurrenceEditor compliant with Section 508 and WCAG 2.0

### ISSUE THREAD

**Issue by veselints** [telerik/kendo-ui-core#4160]
### Bug report
Make Scheduler RecurrenceEditor Section 508 and WCAG 2.0 compliant. Currently, the screen-readers do not announce most of the labels on the inputs in the RecurrenceEditor. For some of the others (the weekday checkboxes for example), only the abbreviation of the day is announced.

**Comment by mtavares628** [comment]
Is there any movement on this?  I brought this issue to your attention over a year and half ago.

**Comment by veselints** [comment]
Implemented in branch `recurrence-editor`. Please do not merge it as I have to drop one of the commits before that.
